compiler/sem/semdata.nim

```diff
@@ -59,9 +59,8 @@ type
 
   POptionEntry* = ref TOptionEntry
   PProcCon* = ref TProcCon
-  # TODO: This type should be marked acyclic
-  TProcCon* = object ## procedure context; also used for top-level
-                                 ## statements
+  TProcCon* {.acyclic.} = object
+    ## procedure context; also used for top-level statements
```

Comment:
You should add a reason for why this is acyclic.

Reply:
Hm, why? The `next` field is explicitly documented as being used for *stacking*.